client/internal/config/config.go

```diff
@@ -0,0 +1,45 @@
+package config
+
+import (
+	"client/internal/logger"
+	"flag"
+	"os"
+)
+
+type MainConfig struct {
+	DatabaseURI string
+	ServerURL   string
+}
+
+func Make() MainConfig {
```

Comment:
 То же самое - не вижу необходимости копировать Config при создании.
```suggestion
func Make() *MainConfig {
```

Reply:
Поправил и в сервере и в клиенте.
Правильно ли я в таком случае понимаю, что всегда, когда возвращается более-менее сложная структура, то нужно возвращать как указатель? Видимо чтобы дальше оно не копировалось при передаче в дочерние структуры.

Reply:
It depends. Я затрагивал эту тему довольно подробно в паре вебинаров. Вообще я руководствуюсь правилом  (и коммьюнити тоже рекомендует так делать) - используй тип указателя, везде, если явно не требуется значение. Значение бывает необходимо передавать во первых, когда мы осознанно хотим чтобы значение копировалось, если мы делаем оптимизацию, например zero-allocation и т.д.